### PR TITLE
 Refactor structure & Developer friendly packages / webpack

### PR DESCRIPTION
**Pull request by Rassell**
Refactor structure  & Developer friendly packages / webpack

**What I did:**
In order to make it more easier to develop some basic restructure should be done, I think container / component is the best approach

**Why I did it:**
As a developer basic structure of a project is very important to understand it and also to start working.
Notice that also a lot of "eslint" (with airbnb configuration I think as a guideline) is needed in order to maintain some code rules / styles

**Fixes issue (include link):**
-

**Mockups, screenshots, evidence:**
-

**Is this related to an open issue?**
-

### CODE REVIEW

**Comment by atomicptr** [comment]
Guess moving everything to src makes sense, since there is no other logic beyond the react app :). I used a template from work where src is usually occupied by the backend logic, but you're right it's better this way.

A few change I'd like to see though:

* Delete view/, I don't quite see why it's still there.
* 404.html should also move into src/, cuz that thing is very important for making Github like SPAs
* Also you should adjust the deploy script in scripts/deploy.sh to cp src/*.html instead of *.html or else changes in the HTML files will no longer be build :)

Besides those points it looks good.

**Comment by Rassell** [comment]
About the view/ it's because was my first time working on github and I forgot/didn't know about "forking" first and was trying to modify this repo 💯 

Now I'm working on setting eslint with the airbnb config / adding jest to run test and later on will focus on new features / resolving bugs 👍

**Comment by atomicptr** [comment]
I've just noticed that an index.html is created in dist/ but the deploy script is still copying the one from src/index.html which results in a non working Github Pages build (since it doesn't include the webpack built js file). You can test this by running the deploy script (but I'd remove the push for testing) and then go into the deploy directory and just use ``http-serve``.

**Comment by Rassell** [comment]
Yes, my bad. The idea it's to let this plugin of webpack do the work:

```js
    plugins: [new HtmlWebPackPlugin({
        template: './src/index.html',
    })],
```

It will inject the tag to the html on build and when we are developing, ex on build for prod:

```html
...
</script>
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" integrity="sha384-DNOHZ68U8hZfKXOrtjWvjxusGo9WQnrNx2sqG0tfsghAvtVlRW3tvkXWZh58N9jp" crossorigin="anonymous">
<script type="text/javascript" src="dauntless-builder.js"></script>
```
I will investigate to make index.html as 404 also so we only use one html file.

**Comment by atomicptr** [comment]
Don't forget that this is run by Github Pages, which means you can't rely on anything beyond pure file delivery on their end. The 404.html is important because Github Pages isn't designed with SPAs in mind, meaning if I go to a page via /b/buildId it will run the 404.html file which has a script in it that redirects you to the index.html containing the original url path. Removing the 404.html isn't an option and the script tag has to be in there at build time (so travis can push it to the gh-pages branch properly)

**Comment by Rassell** [comment]
Didn't know that was running on Github Pages, let me solve that.

**Comment by Rassell** [comment]
OK, didn't know about the GitHub Pages hosting, then I wont touch the 404 but index.html on dist with the script tag should be fine as it have it after build (before deployment) 👍

**Comment by Rassell** [comment]
updated with the latest changes, working atm on eslint, after that I will try to refactor some parts of the app that have duplicated code like ModalItemListItemComponent.jsx

**Comment by atomicptr** [comment]
If you do other things please make seperate PRs for them. All that this PR currently does is adding webpack-dev-server (which also seems to be deprecated?) and html webpack plugin thingy, right?

**Comment by Rassell** [comment]
Yes i will create a separate pr, no worries

Mmmmm maybe the version is deprecated, but the package itself it's not... 

Also some configuration to the webpack to use maps to make more easier the debugging on dev tools but yeah more or less the other things you already make it to src so there is no too much changes 👍

**Comment by atomicptr** [comment]
image

Well I kinda read this as "is deprecated". The maps would be def. useful tho.

**Comment by Rassell** [comment]
you are right sir, didn't check the repo from long time ago (always working with it and on the webpack documentation says nothing about that) moving to webpack-serve

**Comment by Rassell** [comment]
btw I recommend you jest for unit test:

https://medium.com/airbnb-engineering/unlocking-test-performance-migrating-from-mocha-to-jest-2796c508ec50

File: webpack.config.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const path = require("path");
 const HtmlWebPackPlugin = require('html-webpack-plugin');
 
 module.exports = {
+    mode: process.argv.indexOf("--production") > -1 ? "production" : "development",
```
Comment (atomicptr) [inline webpack.config.js]: Why have you readded this?

File: package.json [hunk 2]
```diff
@@ -27,14 +27,15 @@
...
     "glob": "^7.1.2",
+    "html-webpack-plugin": "^3.2.0",
     "js-yaml": "^3.12.0",
     "node-sass": "^4.9.0",
     "sass-loader": "^7.0.3",
     "style-loader": "^0.21.0",
     "webpack": "^4.12.0",
     "webpack-cli": "^3.0.4",
-    "webpack-dev-server": "^3.1.4"
+    "webpack-dev-server": "^3.1.4",
```
Comment (atomicptr) [inline package.json]: Still a dependency ;)?

File: package.json [hunk 3]
```diff
@@ -27,14 +27,15 @@
     "babel-loader": "^7.1.4",
     "babel-preset-react": "^6.24.1",
     "css-loader": "^0.28.11",
-    "html-webpack-plugin": "^3.2.0",
     "glob": "^7.1.2",
+    "html-webpack-plugin": "^3.2.0",
```
Comment (atomicptr) [inline package.json]: Have you figured out a way to use this without breaking the Github Pages?

**Review by atomicptr — CHANGES_REQUESTED** [review]

**Comment by atomicptr** [comment]
Dunno if changing to jest makes a huge difference for the <10 tests we have :). Will read that article tho.

File: webpack.config.js (hunk 1 above)
Comment (Rassell) [reply webpack.config.js]: Its needed for webpack-serve as it doesnt have a dev flag to pass to webpack

**Comment by atomicptr** [comment]
Huh? Reason why you closed it?

**Pull request closed without merging**